Sources/Controllers/TargetMenu/POI/OAPOIViewController.mm

```diff
@@ -484,97 +546,230 @@ - (void) buildRows:(NSMutableArray<OARowInfo *> *)rows
+                                             textPrefix:nil
+                                                   text:link
+                                              textColor:UIColorFromRGB(kHyperlinkColor)
+                                                 isText:YES
+                                              needLinks:YES
+                                                  order:10000
+                                               typeName:nil
+                                          isPhoneNumber:NO
+                                                  isUrl:YES];
+        [rows addObject:row];
+    }
+}
+
+- (NSString *)getSocialMediaUrl:(NSString *)key value:(NSString *)value
+{
+    // Remove leading and closing slashes
+    NSMutableString *sb = [NSMutableString stringWithString:[value trim]];
+    if ([sb characterAtIndex:0] == '/')
+        [sb deleteCharactersInRange:NSMakeRange(0, 1)];
+    NSInteger lastIdx = sb.length - 1;
+    if ([sb characterAtIndex:lastIdx] == '/')
+        [sb deleteCharactersInRange:NSMakeRange(lastIdx, 1)];
+
+    // It cannot be username
+    if ([sb containsString:@"/"])
+        return [@"https://" stringByAppendingString:value];
+
+    NSMutableDictionary<NSString *, NSString *> *urls = [NSMutableDictionary dictionary];
+    urls[@"facebook"] = @"https://facebook.com/";
+    urls[@"vk"] = @"https://vk.com/";
+    urls[@"instagram"] = @"https://instagram.com/";
+    urls[@"twitter"] = @"https://twitter.com/";
+    urls[@"ok"] = @"https://ok.ru/";
+    urls[@"telegram"] = @"https://t.me/";
+    urls[@"flickr"] = @"https://flickr.com/";
+
+    if ([urls.allKeys containsObject:key])
+        return [urls[key] stringByAppendingString:value];
+    else
+        return nil;
+}
+
+- (NSArray<NSString *> *)getFormattedPrefixAndText:(NSString *)key
+                                            prefix:(NSString *)prefix
+                                             value:(NSString *)value
+                                           amenity:(OAPOI *)amenity
+{
+    NSNumberFormatter *numberFormatter = [[NSNumberFormatter alloc] init];
+    [numberFormatter setFormatterBehavior:NSNumberFormatterBehavior10_4];
+    [numberFormatter setNumberStyle:NSNumberFormatterDecimalStyle];
+    numberFormatter.maximumFractionDigits = 2;
+    numberFormatter.decimalSeparator = @".";
+
+    EOAMetricsConstant metricSystem = [[OAAppSettings sharedManager].metricSystem get];
+
+    NSString *formattedValue = value;
+    NSString *formattedPrefix = prefix;
+    if ([key isEqualToString:@"width"])
+    {
+        formattedPrefix = OALocalizedString(@"shared_string_width");
+    }
+    else if ([key isEqualToString:@"height"])
+    {
+        formattedPrefix = OALocalizedString(@"shared_string_height");
+    }
+    else if (([key isEqualToString:@"depth"] || [key isEqualToString:@"seamark_height"]) && [self isNumericValue:value])
+    {
+        double valueAsDouble = [value doubleValue];
```

Comment:
Wouldn't it be easier to call an existing method for distance instead of copy/pasting?